src/lightning/pytorch/loops/loop.py

```diff
@@ -38,6 +39,11 @@ def restarting(self, restarting: bool) -> None:
             if isinstance(loop, _Loop):
                 loop.restarting = restarting
 
+    @property
+    def is_resuming(self) -> bool:
+        """Whether we're resuming training from a checkpoint."""
+        return self._resuming_from_checkpoint
+
```

Comment:
This variable `_resuming_from_checkpoint` feels like an extra. It seems varibale `_loaded_from_state_dict`also gives the saem purpose.

This variable `_resuming_from_checkpoint` seems a bit redundant. From the naming and context, it looks like `_loaded_from_state_dict` might already be serving a similar purpose.  wdyt @sudiptob2  ?

cc: @Borda

Reply:
Hey @bhimrazy, thanks for taking a look at the PR. In my initial implementation, I had reused `_loaded_from_state_dict`. However, as @Borda suggested,  a new variable would be useful to describe the status of this **resuming** case reliably.
Please see the review comments in this thread: https://github.com/Lightning-AI/pytorch-lightning/pull/20775#discussion_r2131892490

Reply:
Oh, okay. Thanks for letting me know.